File: engine/src/main/kotlin/dev/stopptanz/engine/StopInterval.kt
package dev.stopptanz.engine

data class StopInterval(val minMillis: Long, val maxMillis: Long) {
    init {
        require(minMillis >= 0) { "minMillis must be >= 0" }
        require(maxMillis >= minMillis) { "maxMillis must be >= minMillis" }
    }

    companion object {
        /** Placeholder for callers that don't yet trigger auto Stops (no Stop Interval UI/timer wired up). */
        val unused = StopInterval(0, 0)
    }
}

File: web/src/engine/stopInterval.ts
export interface StopInterval {
  minMillis: number
  maxMillis: number
}

export function createStopInterval(minMillis: number, maxMillis: number): StopInterval {
  if (minMillis < 0) throw new Error('minMillis must be >= 0')
  if (maxMillis < minMillis) throw new Error('maxMillis must be >= minMillis')
  return { minMillis, maxMillis }
}

/** Placeholder for callers that don't yet trigger auto Stops (no Stop Interval UI/timer wired up). */
export const unusedStopInterval: StopInterval = { minMillis: 0, maxMillis: 0 }
